lib/view_model/after_auth_view_models/add_post_view_models/add_post_view_model.dart

```diff
@@ -154,4 +164,121 @@ class AddPostViewModel extends BaseModel {
     _imageFile = null;
     notifyListeners();
   }
+
+  /// This function will return the start and end index of the current tag where the pointer cursor is currently placed.
+  ///
+  /// **params**:
+  ///   None
+  ///
+  /// **returns**:
+  /// * `MapEntry<int, int>`: Getting the map.
+  MapEntry<int, int> _getCurrentString() {
```

Comment:
Why have you used a MapEntry? Think of using a List or a tuple kind of data structure instead.

Reply:
Dart doesn't have a built in tuple Data structure and if I have to use it I have to get certain packages. using list for this purpose can make the code more complex that's why I chose Map Entry and taking its key as first value and "value" as the second value.

Reply:
The point is that the use of MapEntry is not semantically correct, as you might think too. By tuple or List, I meant 'some sort of structure that mimics a tuple'. Have a look at Offset maybe, as its use case is also related to positions in fields.

Reply:
Ok I'll check it out!!

Reply:
@literalEval Now I have created my own class to have a tuple like functionality and removed Map Entry

Reply:
Hmm. Seems good. Maybe refactor that out in any other file?

Reply:
I didn't get it What do you mean?